contracts/src/PDPService.sol

```diff
@@ -84,13 +88,18 @@ contract PDPService {
     // proofset owner has exclusive permission to add and remove roots and delete the proof set
     mapping(uint256 => address) proofSetOwner;
     mapping(uint256 => address) proofSetProposedOwner;
-    
 
     // Methods
-    constructor(uint256 _challengeFinality) {
+    // Note: no constructor, use initialize()
```

Comment:
Consider retaining a constructor that prevents initialisation of this contract directly (the initializer should be invoked on the proxy, not the impl).

```
/// @custom:oz-upgrades-unsafe-allow constructor
constructor() {
    _disableInitializers();
}
```

You may then need an `initializeForTest` initializer, or to use a proxy in the tests too.